### ISSUE TITLE
Data and configuration files required for unit and integration tests

### ISSUE THREAD

**Issue by GernotMaier** [gammasim/gammasim-tools#275]
Some of the files used in unit and integrations tests can be retrieved from the databases and are not required to be in the tests/resources directory.

## Files required for unittests:

These files are probably not in the DB:

- [ ]  `tests/unit_tests/test_psf_analysis.py` : `tests/resources/photons-LST-d12.0-za20.0-off0.000_lst_integral.lis`
- [ ]  corsikaConfigTest.yml
- [ ]  (all sim_telarray files *.zstd)

These files are in the DB?

- [ ] tests/unit_tests/test_ray_tracing.py::test_config_data
	- parValues-Sites.yml could not be found
	- parValues-MST-optics.yml could not be found
	- parValues-MST-FlashCam.yml could not be found
	- CTA-North-LST-1-Current_test-telescope-model.cfg
- [ ] tests/unit_tests/test_visualize.py::test_plot_1D - OSError: tests/resources/ref_200_1100_190211a.dat not found
- [ ] tests/unit_tests/test_visualize.py - resources/Transmission_Spectrum_PlexiGlass.dat not found
- [ ] tests/unit_tests/model/test_mirrors.py::test_read_list - tests/resources/mirror_CTA-LST-flen_grouped.dat not found

Not sure what this file does:

- [ ] test_collect_dict_data.yml 

## Files required for integration tests:

- [ ] tests/integration_tests/test_applications.py::test_applications[tune_psf] PSFcurve_data_v2.txt
- [ ] tests/integration_tests/test_applications.py::test_applications[submit_data_from_external::submit] MLTdata-preproduction.usermeta.yml and MLTdata-preproduction.ecsv (are these in DB? Otherwise I would replace them by two 'dummy' files of similar type)
- [ ] test_applications[submit_data_from_external::submit] - set_MST_mirror_2f_measurements_from_external.config.yml (generic configuration file, we will have them at some point in the DB, but probably not now?)
- [ ] tests/integration_tests/test_applications.py::test_applications[produce_array_config]: arrayConfigTest.yml (generic configuration file, ok?)
- [ ] tests/integration_tests/test_applications.py::test_applications[production]: prodConfigTest.yml (generic configuration file, ok?)

**Comment by GernotMaier** [comment]
> I will try to go one by one first to understand what the file is needed for, is it already in the DB (or an equivalent version of it) and whether we should add it if it isn't. A lot of this can already be done, but I would prefer to get feedback/confirmation first (partly cause anyway it will take a couple of days till I can start with it).
> 
> > These files are probably not in the DB:
> > 
> > * [ ]    `tests/unit_tests/test_psf_analysis.py` : `tests/resources/photons-LST-d12.0-za20.0-off0.000_lst_integral.lis`
> > * [ ]    corsikaConfigTest.yml
> > * [ ]    (all sim_telarray files *.zstd)
> 
> Indeed they are not and I do not think they should be. If they are available right now in `tests/resources/` let's keep them there.

- Good - I've marked those as done.


> 
> > These files are in the DB?
> > 
> > * [ ]   tests/unit_tests/test_ray_tracing.py::test_config_data
> >   
> >   * parValues-Sites.yml could not be found
> >   * parValues-MST-optics.yml could not be found
> >   * parValues-MST-FlashCam.yml could not be found
> 
> No, but they shouldn't be, they contain values which are in the DB. I also don't see a direct call for these files in that test function. Do you get an error when you run it (I can try to reproduce by not allowing a DB connection, but I ask anyway).

Try and test:
```
pytest --no-cov tests/unit_tests/test_ray_tracing.py::test_config_data
```

What I don't see, is [this method](https://github.com/gammasim/gammasim-tools/blob/ed18de64724deec403ed5ee06971c377562331ef/simtools/db_handler.py#L422) searching on disk or in the DB for the yml files? 


> 
> > * CTA-North-LST-1-Current_test-telescope-model.cfg
> 
> This is a kind of mock config file (not really mock, it has real values, but not something we usually use). We can put this file in the DB I think. What do you think?
> 
> > * [ ]   tests/unit_tests/test_visualize.py::test_plot_1D - OSError: tests/resources/ref_200_1100_190211a.dat not found
> > * [ ]   tests/unit_tests/test_visualize.py - resources/Transmission_Spectrum_PlexiGlass.dat not found
> > * [ ]   tests/unit_tests/model/test_mirrors.py::test_read_list - tests/resources/mirror_CTA-LST-flen_grouped.dat not found
> 
> These (or similar files) can be obtained from the DB, I will modify the tests accordingly.

Thanks.


> > Not sure what this file does:
> > 
> > * [ ]   test_collect_dict_data.yml
> 
> This is used in a general test but it's very generic. I think it's better to simply write out that file to temporary directory as part of the test and remove it after test ends. I will try to implement that.

Do we need to have them on disk actually for the test, or could we simply write this dict into a temporary file before the tests? Same for tests/resources/test_parameters.yml.


> 
> > ## Files required for integration tests:
> > 
> > * [ ]   tests/integration_tests/test_applications.py::test_applications[tune_psf] PSFcurve_data_v2.txt
> 
> I think this should go into a testing area in the DB for now (in the future it will be permanent). Agree?

Agree.

> 
> > * [ ]   tests/integration_tests/test_applications.py::test_applications[submit_data_from_external::submit] MLTdata-preproduction.usermeta.yml and MLTdata-preproduction.ecsv (are these in DB? Otherwise I would replace them by two 'dummy' files of similar type)
> 
> These are not in the DB. Dummy versions are good and can be added to the test area in the DB mentioned above.

How do we prepare dummy versions? Simply take the example of the MST, or should we randomize the data so that it is truly dummpy?

> 
> > * [ ]   test_applications[submit_data_from_external::submit] - set_MST_mirror_2f_measurements_from_external.config.yml (generic configuration file, we will have them at some point in the DB, but probably not now?)
> 
> Same as above, can be in a temp area in the DB.

Good.

> 
> > * [ ]   tests/integration_tests/test_applications.py::test_applications[produce_array_config]: arrayConfigTest.yml (generic configuration file, ok?)
> > * [ ]   tests/integration_tests/test_applications.py::test_applications[production]: prodConfigTest.yml (generic configuration file, ok?)
> 
> These are necessary I think. Probably best to keep them in the resources directory.

Agree.

**Comment by GernotMaier** [comment]
The only remaining is the MST example? Yes, this is ok and please upload it to the test-data DB.

**Comment by orelgueta** [comment]
The MST example and the PSF test data to upload to the DB (I think I will change the file to use the new data).

Otherwise I didn't mention but I think set_MST_mirror_2f_measurements_from_external.config.yml should stay in the resources directory since it's a config file, not test data (we can open a new DB for those as well if you think it's necessary).

**Comment by GernotMaier** [comment]
Yes, this is good. Nice to see the resource directory cleaned up. We can deal with the config files later.

**Comment by orelgueta** [comment]
Wait, what about MLTdata-preproduction.usermeta.yml? Should it also go into test-data? It's not test data but it's also not a config file.

**Comment by GernotMaier** [comment]
It is metadata directly linked to the test data. MLTdata-preproduction.usermeta.yml and MLTdata-preproduction.ecsv should always come in pairs (I am actually not entirely sure if it shouldn't be even in the header of the ecsv file). I think it should go into test-data.

**Comment by orelgueta** [comment]
OK, uploading it now.

**Comment by orelgueta** [comment]
Completed in #288.
